src/structures/APIMessage.js

```diff
@@ -153,7 +153,9 @@ class APIMessage {
     }
     const embeds = embedLikes.map(e => new MessageEmbed(e).toJSON());
 
-    const components = this.options.components?.map(c => BaseMessageComponent.create(c).toJSON());
+    const components = this.options.components?.map(c =>
+      BaseMessageComponent.create(Array.isArray(c) ? { type: 1, components: c } : c).toJSON(),
```

Comment:
You could import and use `MessageComponentTypes.ACTION_ROW` from the constants file here instead of this magic number.